### PR TITLE
Improved request response

### PR DESCRIPTION
**Pull request by s-furlong**

### CODE REVIEW

File: ServerSpec/http_server_spec/features/support/response.rb [hunk 1]
```diff
@@ -6,6 +6,9 @@ def initialize(response)
 
   def allowed_headers
     response.headers["allow"].split(/[ \t]*,[ \t]*/)
+  rescue
+    require "pry"
```
Comment (twopoint718) [inline ServerSpec/http_server_spec/features/support/response.rb]: I know we put these in to try and debug issues. Be sure to remove them before finishing off this PR. Just remove the whole `rescue` block.

File: src/main/java/Interfaces/InputOutputInterfaces.java [hunk 2]
```diff
@@ -6,14 +6,17 @@
 public interface InputOutputInterfaces {
     void createInputStream(Socket clientSocket) throws IOException;
 
+
```
Comment (twopoint718) [inline src/main/java/Interfaces/InputOutputInterfaces.java]: Try running a source code formatter in IntelliJ IDEA: https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html

File: src/main/java/wrappers/SocketWrappers.java [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 import Interfaces.SocketInterfaces;
+
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.net.Socket;
 
 public class SocketWrappers implements SocketInterfaces {
     private ServerSocket serverSocket;
 
+    public SocketWrappers() {
```
Comment (twopoint718) [inline src/main/java/wrappers/SocketWrappers.java]: You can leave out empty constructors like this. Java automatically inserts something like this for you.

File: src/test/java/mocks/MockInputOutWrapper.java [hunk 4]
```diff
@@ -47,14 +58,19 @@ public void closeInputOutputStreams() {
 
     }
 
+    @Override
+    public void setClientSocket(Socket clientSocket) {
```
Comment (twopoint718) [inline src/test/java/mocks/MockInputOutWrapper.java]: Likewise, I think you can leave this out.

**Review by twopoint718 — CHANGES_REQUESTED** [review]
Just a few cleanup sort of tasks left. I think the IDE can probably do them for you.

Congrats on getting though the server spec! /cc @breadoliveoilsalt @Cristian-Baeza

congratulations-congrats

File: ServerSpec/http_server_spec/features/support/response.rb (hunk 1 above)
Comment (s-furlong) [reply ServerSpec/http_server_spec/features/support/response.rb]: removed

File: src/main/java/Interfaces/InputOutputInterfaces.java (hunk 2 above)
Comment (s-furlong) [reply src/main/java/Interfaces/InputOutputInterfaces.java]: ran reformat for the repository using the IntelliJ intructions

File: src/main/java/wrappers/SocketWrappers.java (hunk 3 above)
Comment (s-furlong) [reply src/main/java/wrappers/SocketWrappers.java]: deleted the constructor

File: src/test/java/mocks/MockInputOutWrapper.java (hunk 4 above)
Comment (s-furlong) [reply src/test/java/mocks/MockInputOutWrapper.java]: removed

**Review by twopoint718 — APPROVED** [review]

**Pull request merged**